onnxruntime/core/providers/rocm/tunable/gemm_hipblaslt.h

```diff
@@ -0,0 +1,173 @@
+    int64_t ldc = params->n;
+    int64_t stride_a = params->n * params->k;
+    int64_t stride_b = params->k * params->m;
+    int64_t stride_c = params->n * params->m;
+    float alpha = static_cast<float>(params->alpha);
+    float beta = static_cast<float>(params->beta);
+    int row_a, col_a, row_b, col_b, row_c, col_c;
+    row_a = lda;
+    col_a = (params->opb == BlasOp::N) ? params->k : params->n;
+    row_b = ldb;
+    col_b = (params->opa == BlasOp::N) ? params->m : params->k;
+    row_c = ldc;
+    col_c = params->m;
+
+    hipblasDatatype_t in_out_datatype = HipBlasDataTypeFor(params->a);
+    hipblasLtMatrixLayout_t mat_a, mat_b, mat_c;
+    hipblasLtMatmulDesc_t matmul;
+    HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutCreate(&mat_a, in_out_datatype, row_a, col_a, lda));
+    HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutCreate(&mat_b, in_out_datatype, row_b, col_b, ldb));
+    HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutCreate(&mat_c, in_out_datatype, row_c, col_c, ldc));
+    HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescCreate(&matmul, HIPBLASLT_COMPUTE_F32, HIPBLAS_R_32F));
+
+    if (params->batch > 1) {
+      int batch_count = params->batch;
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+          mat_a, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &batch_count, sizeof(batch_count)));
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+          mat_a, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &stride_a, sizeof(stride_a)));
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+          mat_b, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &batch_count, sizeof(batch_count)));
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+          mat_b, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &stride_b, sizeof(stride_b)));
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+          mat_c, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &batch_count, sizeof(batch_count)));
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+          mat_c, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &stride_c, sizeof(stride_c)));
+    }
+
+    hipblasOperation_t trans_a = (params->opb == BlasOp::N) ? HIPBLAS_OP_N : HIPBLAS_OP_T;
+    hipblasOperation_t trans_b = (params->opa == BlasOp::N) ? HIPBLAS_OP_N : HIPBLAS_OP_T;
+
+    HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescSetAttribute(
+      matmul, HIPBLASLT_MATMUL_DESC_TRANSA, &trans_a, sizeof(int32_t)));
+    HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescSetAttribute(
+      matmul, HIPBLASLT_MATMUL_DESC_TRANSB, &trans_b, sizeof(int32_t)));
+
+    hipblasLtEpilogue_t epilogue;
+    HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescSetAttribute(
+      matmul, HIPBLASLT_MATMUL_DESC_EPILOGUE, &epilogue, sizeof(epilogue)));
+    hipblasLtMatmulPreference_t pref;
+    void* workspace;
+    HIP_RETURN_IF_ERROR(hipMalloc(&workspace, workspace_size_));
```

Comment:
I think you should use `hipMallocAsync`, also `hipFreeAsync`.

Reply:
Corrected.